libraries/chain/asset_evaluator.cpp

```diff
@@ -607,12 +607,20 @@ void_result asset_publish_feeds_evaluator::do_evaluate(const asset_publish_feed_
       FC_ASSERT( !bitasset.has_settlement(), "No further feeds may be published after a settlement event" );
    }
 
+   // the settlement price must be quoted in terms of the backing asset
    FC_ASSERT( o.feed.settlement_price.quote.asset_id == bitasset.options.short_backing_asset );
+
    if( d.head_block_time() > HARDFORK_480_TIME )
    {
       if( !o.feed.core_exchange_rate.is_null() )
       {
-         FC_ASSERT( o.feed.core_exchange_rate.quote.asset_id == asset_id_type() );
+         if (bitasset.options.short_backing_asset != asset_id_type()) {
+            // on a non-CORE backed asset, the quote asset cannot be null
+            FC_ASSERT( o.feed.core_exchange_rate.quote.asset_id != asset_id_type() );
+         } else {
+            // on a CORE backed asset, the quote should not have an asset.
+            FC_ASSERT( o.feed.core_exchange_rate.quote.asset_id == asset_id_type() );
+         }
```

Comment:
This needs a hard look. I do not believe the former line was correctly handling non-CORE backed assets correctly. My tests now cover both CORE backed and non-CORE backed, and seem to be working fine. But I would like a second set of eyes to make sure the logic is correct, and if perhaps it could be simplified.

One possible simplification: Shouldn't the short_backing_asset always match the core_exchange_rate.quote.asset_id?

Reply:
`settlement_price` should be in forms of `asset:backing_asset`, but `core_exchange_rate` should be always in form of `asset:core`. The code above was related to a hard fork that fixed an old bug (HARDFORK_480). More info: https://github.com/cryptonomex/graphene/issues/480.

For testing new code, we should proceed to a time after that hard fork time first, something like this: https://github.com/bitshares/bitshares-core/blob/233be14666352735d598d1119bd154ce21e50476/tests/tests/settle_tests.cpp#L38-L42

Reply:
TL;DR: don't modify this code block.

Reply:
Reverted changes I had made in this code block to their original, and modified test cases to set core_exchange_rate based on asset:core.